csrc/device_lower/pass/index.cpp

```diff
@@ -405,6 +405,23 @@ void IndexLowering::handle(const TopKOp* top) {
   GpuLower::current()->propagateExprInfo(top, back());
 }
 
+void IndexLowering::handle(const BlockQuantizationOp* bqop) {
+  const auto in = lowerSrcIndex(bqop->in(), bqop->quantizedOutput());
+
+  // For the two outputs, we don't really need indices.
+  const auto out_scales = IrBuilder::create<kir::TensorIndex>(
+      static_cast<TensorView*>(bqop->blockScales()),
+      IrBuilder::create<Val>(0L, DataType::Index));
+
+  const auto out_quantized = IrBuilder::create<kir::TensorIndex>(
+      static_cast<TensorView*>(bqop->quantizedOutput()),
+      IrBuilder::create<Val>(0L, DataType::Index));
+
+  pushBack(
+      IrBuilder::create<BlockQuantizationOp>(out_scales, out_quantized, in));
```

Comment:
I don't see how the runtime function resolves indexing.

i.e. here we use lowerSrcIndex to index the input TV. but passing only the base pointer of outputs. We are not providing any other `index` logic to the runtime function, how are we going to figure out how the input is indexed and map them to the right element?

I felt like we are putting too much faith on the schedule....

Reply:
Thanks, after modifications, this is the generated kernel:
https://gist.github.com/protonu/cdc348769bf79e2ebf024ce02ae5e63a

Reply:
I'll create an issue to track this.

Reply:
I said it doesn't need to be this PR but please consider it would be a blocker for actual scheduler integration.

Reply:
I have the logic for validation in:
https://github.com/NVIDIA/Fuser/pull/5362

I can move that here.

Reply:
I'll resolve this thread - and we can discuss this in the validation function added.